akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MarshallingDirectivesSpec.scala

```diff
@@ -224,5 +224,11 @@ class MarshallingDirectivesSpec extends RoutingSpec with Inside {
         rejection shouldEqual UnacceptedResponseContentTypeRejection(Set(ContentType(`application/json`)))
       }
     }
+    "render JSON response when `Accept` header is present with charset UTF-8 explicitly specified" in {
```

Comment:
Well it's a bit harder because the first obvious case is when `charset` in Accept header is non-UTF. And here we come to the point that Accept header in the akka-http model has no typed method to distinguish charsets (like `HttpCharsetRange` in the `Accept-Charset`). 
Moreover, I've found another failing case: when Accept-Charset is Unicode but not UTF-8 (UTF-16 for example). It should be served as well according to [RFC](https://tools.ietf.org/html/rfc4627#section-3).
Should I try to introduce optional `HttpCharsetRange` param in the `Accept` header class?